### PR TITLE
Adding Antrea specs when CNI variable has antrea as value

### LINKED ISSUE #2582: CNI:none options is not working for Windows clusters
**Bug description**
When using CNI:none and `IS_WINDOWS_WORKLOAD_CLUSTER` option together, all Windows nodes install Antrea and ignore the options passed.

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [ ] Addons
- [ ] CLI
- [ ] Docs
- [ ] IAM
- [x] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Expected behavior**
To have no CNI installed when none is passed

**Steps to reproduce the bug**
Create a Windows cluster and use CNI:none and check Antrea process installation on windows nodes.

**Version** (include the SHA if the version is not obvious)
1.5 and 1.6

### PR DESCRIPTION
**Pull request by knabben**
### What this PR does / why we need it
CNI:none is being ignored on Windows nodes and always install Antrea by default. This PR fix the CNI attribute on windows clusters.

### Which issue(s) this PR fixes

Fixes #2582

### Describe testing done for PR

Testbeds were created with the overlay and confirmed the not installation of the CNI

### Release note

```release-note

```

### PR Checklist

- [ ] Squash the commits into one or a small number of logical commits
      
- [ ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: pkg/v1/providers/ytt/03_customizations/03_windows/register_antrea_cleanup.yaml [hunk 1]
```diff
@@ -1,20 +1,19 @@
...
-#@ if data.values.IS_WINDOWS_WORKLOAD_CLUSTER:
-#@ if data.values.CNI == "antrea":
+#@ if data.values.IS_WINDOWS_WORKLOAD_CLUSTER and data.values.CNI == "antrea":
 #@overlay/match by=overlay.subset({"kind":"KubeadmConfigTemplate"}), expects="1+"
 ---
 spec:
   template:
     spec:
+      #@overlay/match missing_ok=True
       files:
```
Comment (knabben) [inline pkg/v1/providers/ytt/03_customizations/03_windows/register_antrea_cleanup.yaml:12]: do I need to keep `#@overlay/append` here, it seems the default behavior though

Comment (anujc25) [reply pkg/v1/providers/ytt/03_customizations/03_windows/register_antrea_cleanup.yaml:12]: I am not sure. I think we should keep the `append` considering the some version of `ytt` library was replacing `files` with new content. There might be possibility that in some overlay in some other file, we already have some `files` defined and if we are not careful that can get replaced if we don't have `#@overlay/append`

So, my preference is to keep this `#@overlay/append`

**Review by anujc25 — APPROVED** [review]
LGTM.

File: pkg/v1/providers/tests/unit/windows_test.go [hunk 2]
```diff
@@ -50,15 +51,158 @@ var _ = Describe("Windows Ytt Templating", func() {
...
+		Expect(err).NotTo(HaveOccurred())
+
+		compare := reflect.DeepEqual(obj.GetAnnotations(), map[string]string{"osInfo": ",,", "tkg/plan": "prod"})
+		Expect(compare).To(BeTrue())
+	})
+	It("Has correct kubeadmConfigTemplate for CNI:None", func() {
+		values["CNI"] = "none"
+		rawClusterAPIYaml, err := renderYTTTemplate(paths, values)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(rawClusterAPIYaml).To(ContainSubstring("CNI: none"))
```
Comment (randomvariable) [inline pkg/v1/providers/tests/unit/windows_test.go:90]: Can we do better tests than grep please? I've already refactored these tests once.

File: pkg/v1/providers/tests/unit/windows_test.go [hunk 3]
```diff
@@ -67,34 +211,74 @@ var _ = Describe("Windows Ytt Templating", func() {
...
+	}
+	for _, obj := range objs {
+		if kind == obj.GetKind() {
+			unstructObjects = append(unstructObjects, obj)
+		}
+	}
+	return unstructObjects, nil
+}
+
+func searchObjectByName(name, rawYAML string) (*unstructured.Unstructured, error) {
```
Comment (randomvariable) [inline pkg/v1/providers/tests/unit/windows_test.go:239]: Please add function doc header explaining its use.

File: pkg/v1/providers/tests/unit/windows_test.go [hunk 4]
```diff
@@ -67,34 +211,74 @@ var _ = Describe("Windows Ytt Templating", func() {
...
 	}
-
 	for _, obj := range objs {
 		kinds[obj.GetKind()] = kinds[obj.GetKind()] + 1
 	}
 
 	return kinds, nil
 }
 
+func renderYTTTemplate(paths []string, values map[string]interface{}) (string, error) {
```
Comment (randomvariable) [inline pkg/v1/providers/tests/unit/windows_test.go:277]: Add function header

**Comment by knabben** [comment]
This should be supported on ClusterClass clusters at  https://github.com/vmware-tanzu/tanzu-framework/pull/3867

closing this PR in favor of.

**Pull request closed without merging**